lang/cs/Org.Apache.REEF.IO.Tests/TestAzureDataLakeFileSystemE2E.cs

```diff
@@ -0,0 +1,229 @@
+        {
+            // Service principal / appplication authentication with client secret / key
+            // Use the client ID of an existing AAD "Web App" application.
+            // Fill in before running test!
+            const string adlsAccountName = "#####.azuredatalakestore.net";
+            const string tenant = "microsoft.onmicrosoft.com";
+            const string tokenAudience = @"https://datalake.azure.net/";
+            const string clientId = "#############################";
+            const string secretKey = "##########";
+
+            defaultFolderName = "reef-test-folder-" + Guid.NewGuid();
+
+            IConfiguration conf = AzureDataLakeFileSystemConfiguration.ConfigurationModule
+                .Set(AzureDataLakeFileSystemConfiguration.DataLakeStorageAccountName, adlsAccountName)
+                .Set(AzureDataLakeFileSystemConfiguration.Tenant, tenant)
+                .Set(AzureDataLakeFileSystemConfiguration.ClientId, clientId)
+                .Set(AzureDataLakeFileSystemConfiguration.SecretKey, secretKey)
+                .Build();
+
+            _fileSystem = TangFactory.GetTang().NewInjector(conf).GetInstance<AzureDataLakeFileSystem>();
+
+            ServiceClientCredentials adlCreds = GetCreds_SPI_SecretKey(tenant, new Uri(tokenAudience), clientId, secretKey);
+            _adlsClient = AdlsClient.CreateClient(adlsAccountName, adlCreds);
+        }
+
+        public void Dispose()
+        {
+            if (_adlsClient != null)
+            {
+                _adlsClient.DeleteRecursive($"/{defaultFolderName}");
+            }
+        }
+
+        [Fact(Skip = SkipMessage)]
+        public void TestOpenE2E()
+        {
+            string fileName = UploadFromString(ContentsText);
+            var stream = _fileSystem.Open(PathToFile(fileName));
+            StreamReader reader = new StreamReader(stream);
+            string streamText = reader.ReadToEnd();
+            Assert.Equal(ContentsText, streamText);
+        }
+
+        [Fact(Skip = SkipMessage)]
+        public void TestCreateE2E()
+        {
+            string fileName = $"/{defaultFolderName}/TestCreateE2E.txt";
+            var stream = _fileSystem.Create(PathToFile(fileName));
+            Assert.True(_adlsClient.CheckExists(fileName));
+            Assert.Equal(typeof(AdlsOutputStream), stream.GetType());
+        }
+
+        [Fact(Skip = SkipMessage)]
+        public void TestDeleteE2E()
+        {
+            string fileName = UploadFromString(ContentsText);
+            Assert.True(_adlsClient.CheckExists(fileName));
+            _fileSystem.Delete(PathToFile(fileName));
+            Assert.False(_adlsClient.CheckExists(fileName));
+        }
+
+        [Fact(Skip = SkipMessage)]
+        public void TestExistsE2E()
+        {
+            string fileName = UploadFromString(ContentsText);
+            Assert.True(_fileSystem.Exists(PathToFile(fileName)));
+            _adlsClient.Delete(fileName);
```

Comment:
nit: I would use _fileSystem.Delete here.

Reply:
Checking the Exists() method of filesystem here only. I dont think using any other FileSystem method here correct. The setups for the checks should be done by the adlsClient directly